src/pages/mainnet-renderer/[componentAccountId]/widget/[componentName].tsx

```diff
@@ -0,0 +1,77 @@
+import * as nearAPI from 'near-api-js';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+
+import { VmComponent } from '@/components/vm/VmComponent';
+import { useDefaultLayout } from '@/hooks/useLayout';
+import type { NextPageWithLayout } from '@/utils/types';
+
+const ViewComponentPage: NextPageWithLayout = () => {
+  const router = useRouter();
+  const componentSrc = router.query.componentAccountId ? `${router.query.componentAccountId}/widget/${router.query.componentName}` : null;
+  const [ componentCode, setComponentCode ] = useState('return <div>loading mainnet renderer..</div>')
+
+  useEffect(() => {
+    const getComponent = async (componentPath: string | null) => {
+      if(!componentPath) return;
+      const keyPair = nearAPI.KeyPair.fromRandom('ed25519');
+      const keyStore = new nearAPI.keyStores.InMemoryKeyStore();
+      const nearConfig = {
+        networkId: 'testnet',
+        keyStore,
+        nodeUrl: 'https://rpc.testnet.near.org', // todo from env?
+        helperUrl: 'https://helper.testnet.near.org',
+        headers: {},
+      };
+      await keyStore.setKey(
+        nearConfig.networkId,
+        'dontcare',
+        keyPair,
+      );
+
+      const near = await nearAPI.connect(nearConfig);
+      const account = await near.account('dontcare');
+
+      const component = await account.viewFunction({
+        contractId: 'v1.social08.testnet',
+        methodName: 'get',
+        args: {
+          keys: [`${componentPath}/**`],
+        },
+      });
+
+      return componentPath
+        .split('/')
+        .reduce((acc, curr) => acc[curr] || acc, component);
+    }
+    getComponent(componentSrc).then((code) => {
+      if(code && code[""]) {
+        setComponentCode(code[""])
+      } else {
+        setComponentCode('return <div>Not Found</div>')
+      }
+    })
+  }, [componentSrc]);
+
+  return (
+    <div className="container-xl">
+      <div className="row">
+        <div
+          className="d-inline-block position-relative overflow-hidden"
+          style={{
+            paddingTop: 'var(--body-top-padding)',
+          }}
+        >
+          <VmComponent
+            code={componentCode}
```

Comment:
I don't have much context here, but I'm confused why it's necessary to fetch the code manually when the VM itself seems to support passing in a `networkId` per `Widget`? Seems to render the `mainnetWidget` boolean prop meaningless as well.

Reply:
In this case we need to use the `code` prop because if we give the `Widget` component a path with `networkId` set to `mainnet` then it will attempt to resolve that path on `mainnet` SocialDB i.e. `social.near`. The goal here is run code on `testnet` SocialDB i.e. `v1.social08.testnet` but run that code as mainnet code. 

E.g. in [the example](https://near-discovery-git-mainnet-renderer-near-developer-console.vercel.app/mainnet-renderer/dev-1685373830583-51701575881136/widget/NearSocialComponent) I took the devhub code from mainnet SocialDB path `devgovgigs.near/widget/Ideas` as is and copied it to a new path `dev-1685373830583-51701575881136/widget/NearSocialComponent` on testnet SocialDB. The `mainnet-renderer` route fetches the code from the new testnet path and runs it as mainnet code i.e. resolving child `Widget` paths on mainnet SocialDB and Near.view and Social.get and other BOS API usages will be pointing to mainnet.